egs/wsj/s5/utils/create_wdisambig_file.pl

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/perl -w
+# Copyright 2016 FAU Erlangen (Author: Axel Horndasch)
+
+# usage:  create_wdisambig_file.pl category_list.txt words_or_phones.txt disambig_symbol_array.txt
+
+# This script creates a file which is usually called 'wdisambig.txt'. That file
+# contains a (line-by-line) list of the text-form of the disambiguation symbols
+# that are used in the grammar and passed through by the lexicon.
+#
+# See also 'prepare_lang.sh' (look for 'wdisambig.txt').
+use strict;
+use Getopt::Long;
+
+# options
+my $word_class_list     = "";
+my $remove_class_prefix = "";
+
+# helper variables
+my @word_list;
+
+my %optctl = (
+    word_class_list     => \$word_class_list,
+    remove_class_prefix => \$remove_class_prefix,
+    );
+
+# get the option values
+&GetOptions(\%optctl,
+    "word_class_list=s",
+    "remove_class_prefix=s",
+    );
+
+# start the list of disambiguation symbols with #0 (empty word symbol)
+print "#0\n";
+
+if(defined($word_class_list) and $word_class_list ne "") {
```

Comment:
Also note: just because some of my comments are not very minor, does not necessarily mean I am OK with the general structure of this.  At this point it wasn't super clear to me, from looking at your patch, how it works overall, and as a rule I don't spend too much time figuring these things out because if it's not immediately obvious, then the code needs to be explained better or made more obvious.  I'll re-review this after you've made some clarifications and maybe broken some parts of it out into separate scripts (if you think it makes sense).
Think about how much effort it will take a user, from looking at what you've done, to understand how this works, and to implement the same idea in a different setup (say, switchboard).  Mixing things too deeply into the data-prep scripts isn't always a good idea because those are corpus dependent.  Now, the LM training is part of the data-prep so it's hard to separate from the local scripts, but if you could find a way to isolate your changes somehow so that it's easier to apply to other setups, it would make things easier.   E.g. have a step that happens just before building the ARPA LM, and then a later step where you make a modification to the FST after the ARPA is built.  Feel free to define a special directory structure, if it would be helpful, where you store info about the word classes.